pkg/github/client/client.go

```diff
@@ -52,11 +53,59 @@ var runnerPerMinuteRate = map[string]float64{
 
 // New instantiates a new GitHub API client.
 func New(ctx context.Context, settings models.Settings) (*Client, error) {
-	if settings.AccessToken == "" {
-		// If access token is not set, return downstream error as it is required.
-		return nil, errorsource.DownstreamError(fmt.Errorf("access token is required"), false)
+	if settings.PrivateKey != "" {
+		return createAppClient(settings)
 	}
 
+	if settings.AccessToken != "" {
+		return createAccessTokenClient(ctx, settings)
+	}
+
+	return nil, errorsource.DownstreamError(fmt.Errorf("access token or app token are required"), false)
+}
+
+func createAppClient(settings models.Settings) (*Client, error) {
+	appId, err := strconv.ParseInt(settings.AppId, 10, 64)
+	if err != nil {
+		return nil, errorsource.DownstreamError(fmt.Errorf("error parsing app id: %w", err), false)
+	}
+
+	installationId, err := strconv.ParseInt(settings.InstallationId, 10, 64)
+	if err != nil {
+		return nil, errorsource.DownstreamError(fmt.Errorf("error parsing installation id: %w", err), false)
```

Comment:
this is great that we are marking them as downstream error, but I noticed that they are still plugin errors, becuase at the end, we are not returning that error 
 in https://github.com/grafana/github-datasource/blob/main/pkg/plugin/instance.go#L40. We should probably just return full error. Or what do you think?